### PR TITLE
Fixes #22584: "Variable string from command" creates empty files in /var/rudder/modified-files

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/22584

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

File: tree/30_generic_methods/variable_string_from_command.cf [hunk 1]
```diff
@@ -92,6 +92,18 @@ bundle agent variable_string_from_command(prefix, name, command)
       "pass2" expression => "pass1";
       "pass1" expression => "any";
 
+  files:
+    # avoid backups for tmp files
+    !pass2::
+      "${tmp_script_name}"
+        create        => "true",
+        edit_line     => insert_lines("${lines}"),
```
Comment (ncharles) [inline tree/30_generic_methods/variable_string_from_command.cf:100]: i thnk the lines are not valid

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: tree/30_generic_methods/variable_string_from_command.cf [hunk 2]
```diff
@@ -78,31 +78,40 @@ bundle agent variable_string_from_command(prefix, name, command)
...
+    # avoid backups for tmp files
+    pass1.!pass2::
+      "${tmp_script_name}"
+        create        => "true",
+        edit_line     => insert_lines("${command}"),
+        edit_defaults => empty,
+        perms         => mog("700", "root", "0"),
+        comment       => "${class_prefix}",
+        action        => immediate_ignore_dry_run,
+        classes       => classes_generic("file_lines_present_${c_tmp_script_name}");
```
Comment (ncharles) [inline tree/30_generic_methods/variable_string_from_command.cf:105]: you could even use  "inner_class_prefix", to remove the name of the method we don't use anymore

Comment (Fdall) [inline tree/30_generic_methods/variable_string_from_command.cf:105]:
```suggestion
        classes       => classes_generic("${inner_class_prefix}");
```

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**